lab_5/work_with_db.py

```diff
@@ -0,0 +1,511 @@
+    bank_serving_oldest_client = account_of_oldest_user['bank_id']
+
+    return bank_serving_oldest_client
+
+
+def get_bank_with_most_unique_outbound_transactions():
+    """
+    Retrieves the bank with the highest number of unique users who performed outbound transactions.
+
+    :return: The name of the bank with the most unique users performing outbound transactions or None if no banks
+    are found.
+    """
+    transactions = get_data_from_db('Transactions')
+    banks = {}
+    for transaction in transactions:
+        if transaction['bank_sender_name'] not in banks:
+            banks[transaction['bank_sender_name']] = set()
+        banks[transaction['bank_sender_name']].add(transaction['account_sender_id'])
+
+    bank_with_most_unique_users = max(banks, key=lambda bank: len(banks[bank]))
+    return bank_with_most_unique_users
+
+
+def delete_row(data_to_delete, del_user):
+    """
+    Deletes either user or account rows based on the input data.
+
+    :param data_to_delete: A list of user IDs or account IDs to be deleted.
+    :param del_user: If True, delete user rows; if False, delete account rows.
+    """
+    [delete_from_db(i, del_user) for i in data_to_delete]
+    logging.debug('Deleted data: %s', data_to_delete)
+
+
+def delete_users_and_accounts_with_missing_info():
+    """
+    Deletes users and accounts with missing information (NULL values).
+    """
+    users = get_data_from_db('Users')
+    accounts = get_data_from_db('Accounts')
+    users_to_delete = [user for user in users if None in user.values()]
+    accounts_to_delete = [account for account in accounts if None in account.values()]
+
+    [delete_row(user['id'], 'User') for user in users_to_delete]
+    [delete_row(account['id'], 'Accounts') for account in accounts_to_delete]
+
+    logging.info('Users and accounts data without all information has been successfully deleted.')
+
+
+def convert_transactions_datetime(transactions):
+    """
+    Converts the datetime strings in a list of transactions to datetime objects.
+
+    :param transactions: A list of transactions.
+    :return: The same list of transactions, but with datetime strings replaced by datetime objects.
+    """
+    return [{**transaction, 'datetime': datetime.strptime(transaction['datetime'], '%Y-%m-%d %H:%M:%S')}
+            for transaction in transactions]
+
+
+def filter_transactions_past_3_months(user_id):
+    """
+    Filters transactions for the past 3 months.
+
+    :param user_id: The ID of the user whose transactions to filter.
+    :return: A list of transactions for the user in the past 3 months.
+    """
+    transactions = get_transactions_for_user(user_id)
+
+    three_months_ago = datetime.now() - timedelta(days=90)
+    filtered_transactions = [transaction for transaction in transactions if transaction['datetime'] >= three_months_ago]
+
+    return filtered_transactions
```

Comment:
transactions = get_transactions_for_user(user_id)
    three_months_ago = datetime.now() - timedelta(days=90)
    return  [transaction for transaction in transactions if transaction['datetime'] >= three_months_ago]

What do you think?